aggregator/src/aggregator/aggregation_job_driver.rs

```diff
@@ -1895,6 +1895,314 @@ mod tests {
         assert_eq!(want_batch, got_batch);
     }
 
+    #[tokio::test]
+    async fn step_time_interval_aggregation_job_init_partially_garbage_collected() {
+        // This is a regression test for https://github.com/divviup/janus/issues/2464.
+        const OLDEST_ALLOWED_REPORT_TIMESTAMP: Time = Time::from_seconds_since_epoch(1000);
+        const REPORT_EXPIRY_AGE: Duration = Duration::from_seconds(500);
+        const TIME_PRECISION: Duration = Duration::from_seconds(10);
+
+        // Setup: insert an "old" and "new" client report, and add them to a new aggregation job.
+        install_test_trace_subscriber();
+        let mut server = mockito::Server::new_async().await;
+        let clock = MockClock::new(OLDEST_ALLOWED_REPORT_TIMESTAMP);
+        let ephemeral_datastore = ephemeral_datastore().await;
+        let ds = Arc::new(ephemeral_datastore.datastore(clock.clone()).await);
+        let vdaf = Arc::new(Prio3::new_count(2).unwrap());
+
+        let task = TaskBuilder::new(QueryType::TimeInterval, VdafInstance::Prio3Count)
+            .with_helper_aggregator_endpoint(server.url().parse().unwrap())
+            .with_report_expiry_age(Some(REPORT_EXPIRY_AGE))
+            .with_time_precision(TIME_PRECISION)
+            .build();
+
+        let leader_task = task.leader_view().unwrap();
+
+        let first_time = OLDEST_ALLOWED_REPORT_TIMESTAMP
+            .sub(&Duration::from_seconds(3 * TIME_PRECISION.as_seconds()))
+            .unwrap()
+            .to_batch_interval_start(&TIME_PRECISION)
+            .unwrap();
+        let first_batch_identifier =
+            TimeInterval::to_batch_identifier(&leader_task, &(), &first_time).unwrap();
+        let first_report_metadata = ReportMetadata::new(random(), first_time);
```

Comment:
nit: rename `first_time`, `first_batch_identifier` and `first_report_metadata` to indicate the intent, making it more clear to future test maintainers what condition this exercises? For instance, I think the point here is that the `first_*` variables pertain to a GC eligible report while the `second_*` ones are for an uneligible report. So `gc_eligible_time`, `gc_eligible_report_metadata` and `gc_uneligible_time`, `gc_uneligible_report_metadata`, perhaps?